quiche/src/lib.rs

```diff
@@ -2071,6 +2081,27 @@ impl Connection {
         Ok(())
     }
 
+    /// Returns true if a QUIC packet is a stateless reset.
+    fn is_stateless_reset(&self, buf: &[u8]) -> bool {
+        // If the packet is too small, then we just throw it away
+        let buf_len = buf.len();
+        if buf_len < 21 {
+            return false;
+        }
+
+        match self.peer_transport_params.stateless_reset_token {
+            Some(token) => {
+                let token_len = 16;
+                if let Ok(token_from_buf) = buf[buf_len - token_len..buf_len].try_into() {
+                    return token == u128::from_be_bytes(token_from_buf)
```

Comment:
RFC9000 calls for a constant-time comparison of the reset token (see https://www.rfc-editor.org/rfc/rfc9000#section-10.3.1-4) and I don't think it's the case here.

The ring crate we already depend on provides a constant-time comparison API that could be used here, see https://docs.rs/ring/0.16.20/ring/constant_time/fn.verify_slices_are_equal.html